model/src/test/kotlin/config/NexusIqConfigurationTest.kt

```diff
@@ -33,14 +33,14 @@ import org.ossreviewtoolkit.model.writeValue
 class NexusIqConfigurationTest : WordSpec({
     "NexusIqConfiguration" should {
         "support a serialization round-trip via an ObjectMapper" {
-            val referenceOrtConfig = OrtConfiguration.load(file = File("src/main/resources/reference.conf"))
-            val rereadOrtConfig = createTempFile(suffix = ".yml").toFile().apply {
-                writeValue(referenceOrtConfig)
-                deleteOnExit()
-            }.readValue<OrtConfiguration>()
+            val ortConfig = OrtConfiguration.load(file = File("src/main/resources/reference.conf"))
+            val rereadOrtConfig = createTempFile(suffix = ".yml").toFile().run {
+                writeValue(ortConfig)
+                readValue<OrtConfiguration>().also { delete() }
```

Comment:
I am not sure whether this kind of change is safe. If an exception is thrown by `readValue()` or the test fails for other reasons, the temporary file will not be deleted, will it?

It seems that Kotest has a feature to support [temporary files](https://kotest.io/docs/framework/temporary-files/). Would this be an alternative?

Reply:
> It seems that Kotest has a feature to support [temporary files](https://kotest.io/docs/framework/temporary-files/). Would this be an alternative?

That's indeed an alternative I was also looking into, but I didn't want to take it into use until there is a Kotest release which includes [my changes to their directory naming](https://github.com/kotest/kotest/pull/2140). But then again, maybe we should start using the feature already now, and only benefit from the improved naming on the next Kotest upgrade.

Reply:
Uh, I just discovered another issue with Kotest's `tempdir()` implementation that needs to be fixed first...